CRM/CiviGeometry/BAO/Geometry.php

```diff
@@ -265,8 +265,63 @@ public function calculateDistance($params) {
       1 => [$params['geometry_a'], 'String'],
       2 => [$params['geometry_b'], 'String'],
     ]);
-    $meters = (float) $result * 1000;
-    return $meters;
+    $metres = (float) $result * 1000;
+    return $metres;
+  }
+
+  public static function returnSpatialInformation($geometryID) {
+    return CRM_Core_DAO::executeQuery("SELECT
+        id,
+        label
+        , round((ST_Area(geometry)*10000), 3) as square_km
+        , ST_AsText(ST_Envelope(geometry)) as ST_Envelope
+        , ST_AsText(ST_Centroid(geometry)) as ST_Centroid
+        , ST_IsSimple (geometry) as ST_IsSimple
+        , ST_SRID(geometry) as ST_SRID
+      FROM %1
+      WHERE id = %2", [
+      1 => [self::getTableName(), 'MysqlColumnNameOrAlias'],
+      2 => [$geometryID, 'Positive'],
+    ])->fetchAll()[0];
+  }
+
+  /**
+   * Output GeoJSON in a customFormat
+   * @param json $geoJSON - GeoJSON content to be passed out
+   */
+  public static function outputGeometryInFormat($geoJSON, $format) {
+    self::safeLoadAutoloader();
+    $polygon = geoPHP::load($geoJSON, 'json');
+    return $polygon->out($format);
+  }
+
```

Comment:
Haha... this was one of those things where I was wondering if we can get rid of it.  Why do we need envelopes?

Reply:
These boundaries are used for zooming the various interactive maps i believe hence why i have sought to retain them here but we don't need to store the values in a table

Reply:
I guess that kind of denormalisation could be useful if there are some performance gains in terms of producing such a map for the end user.  At least, let's make this denormalisation automatic so that I don't have to do it myself when updating poly boundaries.